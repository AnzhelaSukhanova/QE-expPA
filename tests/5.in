(set-option :produce-models true)
(set-logic BV)
(declare-fun y () (_ BitVec 4))
(declare-fun z () (_ BitVec 4))
(assert
 (exists ((x (_ BitVec 4))) (and 
 (or (bvult x y) (= x y)) 
 (or (bvult z x) (= z x)))))
