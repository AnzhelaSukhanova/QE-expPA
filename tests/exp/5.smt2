(set-option :produce-models true)
(set-logic BV)
(declare-fun y () (_ BitVec 4))
(declare-fun z () (_ BitVec 4))
(assert
 (exists ((x (_ BitVec 4))) (or (bvult (bvshl #b0001 x) (bvadd y (bvadd (bvmul z #b0011) #b1000))) (= (bvshl #b0001 x) (bvadd y (bvadd (bvmul z #b0011) #b1000))))))
