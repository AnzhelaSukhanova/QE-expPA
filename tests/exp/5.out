(let (($e1 (bvadd y (bvadd (bvmul z #b0011) #b1000)))) (let (($e2 (ite (bvult $e1 #b1000) (ite (bvult $e1 #b0100) (ite (bvult $e1 #b0010) (ite (= $e1 #b0000) #b0100 #b0000) #b0001) #b0010) #b0011))) (let (($e3 (bvshl #b0001 $e2))) (let (($e4 (bvadd y (bvadd #b1000 (bvmul z #b1100))))) (let (($e5 (bvshl #b0001 (bvadd #b0001 $e2)))) (let (($e6 (= #b0000 $e4))) (not (and (not (= #b0010 $e4)) (not (= #b0001 $e4)) (not (bvult #b0010 $e4)) (not (bvult #b0001 $e4)) (not (= #b0100 $e4)) (not (bvult #b0100 $e4)) (not (= #b1000 $e4)) (not (bvult #b1000 $e4)) (not (= $e4 $e5)) (not (= $e3 $e4)) (not $e6) (not (bvult $e5 $e4)) (not (bvult $e3 $e4)) $e6))))))))
